python-sdk/nuscenes/nuscenes.py

```diff
@@ -92,6 +92,50 @@ def table_root(self) -> str:
         """ Returns the folder where the tables are stored for the relevant version. """
         return osp.join(self.dataroot, self.version)
 
+    def __getattribute__(self, name):
```

Comment:
why not `__getattr__`?
we can do something like:
```python
def __getattr__(self, name):
    if name == table1_name:
        table = load_table()
        build_index(table)
        setattr(self, name, table)
        return getattr(self, name)
```

Reply:
In general that looks like a good approach, but I think here we want to explicitly store the tables in `self.tables`, so I prefer to explicitly check `if name not in self.tables`.